### PR TITLE
Gen8avc

### PR DESCRIPTION
**Pull request by laureatian**
modify the code according to review advices

### CODE REVIEW

File: src/i965_gpe_utils.c [hunk 1]
```diff
@@ -2663,6 +2663,7 @@ gen9_add_2d_gpe_surface(VADriverContextP ctx,
                         int index)
```
Comment (pengche1) [inline src/i965_gpe_utils.c]: make gpe utils compatable with gen8 --> make gpe utils compitble with gen8

File: src/gen9_avc_encoder.c [hunk 2]
```diff
@@ -1867,6 +2091,48 @@ void gen9_avc_set_image_state(VADriverContextP ctx,
...
+        if (i == 0) {
+            cmd.dw4.macroblock_stat_enable = 0;
+            cmd.dw5.non_first_pass_flag = 0;
+        } else {
+            cmd.dw4.macroblock_stat_enable = 1;
+            cmd.dw5.non_first_pass_flag = 1;
+            cmd.dw5.intra_mb_ipcm_flag = 1;
+            cmd.dw3.inter_mb_conf_flag = 1;
+            cmd.dw3.intra_mb_conf_flag = 1;
+
```
Comment (pengche1) [inline src/gen9_avc_encoder.c]: please delete some unused blank line, make the code clean and beauty

File: src/gen9_avc_encoder.c [hunk 3]
```diff
@@ -2662,14 +2929,23 @@ gen9_avc_send_surface_brc_frame_update(VADriverContextP ctx,
...
+    if (!IS_GEN8(i965->intel.device_info)) {
+        i965_add_buffer_gpe_surface(ctx,
+                                    gpe_context,
+                                    &avc_ctx->res_mb_status_buffer,
+                                    0,
+                                    avc_ctx->res_mb_status_buffer.size,
+                                    0,
+                                    (is_g95 ? GEN95_AVC_FRAME_BRC_UPDATE_MB_STATUS_INDEX : GEN9_AVC_FRAME_BRC_UPDATE_MB_STATUS_INDEX));
+    }
+    if (IS_GEN8(i965->intel.device_info)) {
```
Comment (pengche1) [inline src/gen9_avc_encoder.c]: please use "else" to replace this "if"

File: src/gen9_avc_encoder.c [hunk 4]
```diff
@@ -5550,7 +5830,7 @@ gen9_avc_set_curbe_sfd(VADriverContextP ctx,
     cmd->dw29.vdenc_output_image_state_index = GEN9_AVC_SFD_VDENC_OUTPUT_IMAGE_STATE_INDEX ;
 
     i965_gpe_context_unmap_curbe(gpe_context);
-
+    return;
```
Comment (pengche1) [inline src/gen9_avc_encoder.c]: delete this return

File: src/gen9_avc_encoder.c [hunk 5]
```diff
@@ -5634,15 +5913,1022 @@ gen9_avc_kernel_sfd(VADriverContextP ctx,
...
+
+    } else {
+        cmd->dw2.pitch_width = generic_state->frame_width_in_mbs;
+        cmd->dw4.picture_height_minus1 = generic_state->frame_height_in_mbs - 1;
+        cmd->dw5.slice_mb_height = (avc_state->arbitrary_num_mbs_in_slice) ? generic_state->frame_height_in_mbs : avc_state->slice_height;
+
+        {
+            memcpy(&(cmd->dw8), gen9_avc_mode_mv_cost_table[slice_type_kernel[generic_state->frame_type]][qp], 8 * sizeof(unsigned int));
+            if ((generic_state->frame_type == SLICE_TYPE_I) && avc_state->old_mode_cost_enable) {
+                //cmd.g9->dw8 = gen9_avc_old_intra_mode_cost[qp];
```
Comment (pengche1) [inline src/gen9_avc_encoder.c]: just delete masked codes if it doesn't have some special purpose

File: src/gen9_avc_encoder.c [hunk 6]
```diff
@@ -5634,15 +5913,1022 @@ gen9_avc_kernel_sfd(VADriverContextP ctx,
...
+
+    struct mbenc_param * curbe_param = (struct mbenc_param *)param ;
+    unsigned char qp = 0;
+    unsigned char me_method = 0;
+    unsigned int mbenc_i_frame_dist_in_use = curbe_param->mbenc_i_frame_dist_in_use;
+    unsigned int table_idx = 0;
+    unsigned int curbe_size = 0;
+
+    unsigned int preset = generic_state->preset;
+    if (IS_CHERRYVIEW(i965->intel.device_info) ||
```
Comment (pengche1) [inline src/gen9_avc_encoder.c]: Why just add IS_CHERYVIEW() here?

File: src/gen9_mfc.c [hunk 7]
```diff
@@ -38,6 +38,7 @@
 #include "i965_structs.h"
 #include "i965_drv_video.h"
 #include "i965_encoder.h"
+#include "i965_encoder_api.h"
```
Comment (pengche1) [inline src/gen9_mfc.c]: this commit is only for GEN8, please don't touch GEN9

**Comment by uartie** [comment]
TIP: don't open a new pull request when you need to make changes/additions to an existing pull request. Just push changes to your forked tree/branch associated with the pull-request (e.g. laureatian:gen8avc)... those changes will automatically be reflected in the existing pull-request.  This makes it easier to track the history of a pull-request.

**Comment by laureatian** [comment]
I have updated my repo according to your advices, you can have a look to see if it need more modify?

File: src/gen9_avc_encoder.c [hunk 8]
```diff
@@ -5634,17 +5910,1000 @@ gen9_avc_kernel_sfd(VADriverContextP ctx,
...
+        curbe_cmd->dw10.mbv_proc_states_bti_top_field  = GEN8_SCALING_FIELD_TOP_MBVPROCSTATS_DST_CM;
+        curbe_cmd->dw11.mbv_proc_states_bti_bottom_field = GEN8_SCALING_FIELD_BOT_MBVPROCSTATS_DST_CM;
+    }
+
+    i965_gpe_context_unmap_curbe(gpe_context);
+    return;
+}
+
+static void
+gen8_avc_send_surface_scaling(VADriverContextP ctx,
```
Comment (QuPengfei) [inline src/gen9_avc_encoder.c]: this function is almost same as gen9_avc_send_surface_scaling. is it possible to reuse gen9_avc_send_surface_scaling?

File: src/gen9_avc_encoder.c [hunk 9]
```diff
@@ -5634,17 +5910,1000 @@ gen9_avc_kernel_sfd(VADriverContextP ctx,
...
+    cmd->dw85.brc_curbe_surf_index = GEN8_AVC_MBENC_BRC_CURBE_DATA_CM;
+    cmd->dw86.static_detection_cost_table_index = GEN8_AVC_MBENC_STATIC_FRAME_DETECTION_OUTPUT_CM;
+
+    i965_gpe_context_unmap_curbe(gpe_context);
+
+    return;
+}
+
+static void
+gen8_avc_send_surface_mbenc(VADriverContextP ctx,
```
Comment (QuPengfei) [inline src/gen9_avc_encoder.c]: this function is almost same as gen9_avc_send_surface_mbenc. is it possible to reuse gen9_avc_send_surface_mbenc?

**Comment by laureatian** [comment]
all send surfaces has been reused. @QuPengfei

**Pull request merged**